### PR TITLE
fix(trends): Make breakdowns work with cohorts in poe v2

### PR DESCRIPTION
**Pull request by neilkakkar**
## Problem

Fixes https://posthog.sentry.io/issues/4102203928/?project=1899813&referrer=slack and https://posthog.sentry.io/issues/4099964127/?query=is%3Aunresolved+experiment&referrer=issue-stream&statsPeriod=14d&stream_index=0

@yakkomajuri I decided not to reconcile `PersonPropertiesMode` and `PersonOnEventsMode` yet, because we're not ready for it: The non-poe related things in person properties mode make it pretty hard to try and merge this right now; and leads to discarding support for these things, which is not what we want (like `USING_SUBQUERY` ).

In most cases, there's nothing different in here, but for the few cases where there are differences, I feel this works much better than also introducing checking for poe mode in this filtering, and has the added benefit of fixes in one place translating for all other insight types, because they're handled together

There's another pretty tricky bugfix here:

(1) We use should_join_distinct_ids to determine whether we should join overrides table or not. (and pdi in the non-overrides case)
(2) This gets borked for the `precalculated-cohort` case, because `is_using_cohort_properties` was only checking 'cohort' prop type, not `precalculated-cohort` or `static-cohort`.
(3) Poe V1 doesn't want any joins, even when cohorts are using person properties, so we short-circuit the above evaluation by default.

## Changes

👉 *Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*

## How did you test this code?

Unit tests

### CODE REVIEW

File: posthog/models/property/util.py [hunk 1]
```diff
@@ -319,9 +317,12 @@ def parse_prop_clauses(
 
             method = format_static_cohort_query if prop.type == "static-cohort" else format_precalculated_cohort_query
             filter_query, filter_params = method(cohort, idx, prepend=prepend)  # type: ignore
-            filter_query = f"""{person_id_joined_alias if not person_properties_mode == PersonPropertiesMode.DIRECT_ON_EVENTS else 'person_id'} IN ({filter_query})"""
+            filter_query = f"""{person_id_joined_alias if person_properties_mode not in [PersonPropertiesMode.DIRECT_ON_EVENTS] else 'person_id'} IN ({filter_query})"""
```
Comment (yakkomajuri) [inline posthog/models/property/util.py:320]: nit: do we need a list here?

File: posthog/queries/breakdown_props.py [hunk 2]
```diff
@@ -146,7 +159,10 @@ def get_breakdown_prop_values(
         filter.breakdown_group_type_index,
         filter.hogql_context,
         filter.breakdown_normalize_url,
-        direct_on_events=True if person_properties_mode == PersonPropertiesMode.DIRECT_ON_EVENTS else False,
+        direct_on_events=True
+        if person_properties_mode
+        in [PersonPropertiesMode.DIRECT_ON_EVENTS, PersonPropertiesMode.DIRECT_ON_EVENTS_WITH_POE_V2]
+        else False,
```
Comment (yakkomajuri) [inline posthog/queries/breakdown_props.py:165]: I guess both of us missed this (maybe we both were just using replace tools) but `True if x else False` :DD

**Review by yakkomajuri — APPROVED** [review]

File: posthog/queries/breakdown_props.py (hunk 2 above)
Comment (neilkakkar) [reply posthog/queries/breakdown_props.py:165]: dang it 😂

File: posthog/models/property/util.py (hunk 1 above)
Comment (neilkakkar) [reply posthog/models/property/util.py:320]: nope, will rm

**Pull request merged**